server/src/main/java/org/elasticsearch/common/lucene/search/function/FieldValueFactorFunction.java

```diff
@@ -73,7 +72,7 @@ public double score(int docId, float subQueryScore) throws IOException {
                     if (missing != null) {
                         value = missing;
                     } else {
-                        throw new ElasticsearchException("Missing value for field [" + field + "]");
+                        throw new IllegalArgumentException("Missing value for field [" + field + "]");
```

Comment:
@javanna you might take interest here. This SEEMS like the right thing to do. If a field is missing, it means the data or the query is wrong and the user needs to update, right?

Reply:
I would agree, thanks for the ping